src/utils/daos_metrics/daos_metrics.c

```diff
@@ -53,29 +53,120 @@ print_usage(const char *prog_name)
 	       "\tInclude gauges\n"
 	       "--read, -r\n"
 	       "--reset, -e\n"
+	       "\tReset metrics value to 0\n"
+	       "--jobid, -j\n"
+	       "\tDisplay metrics of the specified job\n"
 	       "\tInclude timestamp of when metric was read\n",
 	       prog_name);
 }
 
-int
-main(int argc, char **argv)
+static int
+process_metrics(int metric_id, char *dirname, int format, int filter,
+		int extra_descriptors, int delay, int num_iter,
+		d_tm_iter_cb_t iter_cb, void *arg)
 {
 	struct d_tm_node_t	*root = NULL;
 	struct d_tm_node_t	*node = NULL;
 	struct d_tm_context	*ctx = NULL;
+	int			iteration = 0;
+	int			rc = 0;
+
+	ctx = d_tm_open(metric_id);
+	if (!ctx)
+		D_GOTO(out, rc = 0);
+
+	root = d_tm_get_root(ctx);
+	if (!root)
+		D_GOTO(out, rc = -DER_NONEXIST);
+
+	if (strncmp(dirname, "/", D_TM_MAX_NAME_LEN) != 0) {
+		node = d_tm_find_metric(ctx, dirname);
+		if (node != NULL) {
+			root = node;
+		} else {
+			printf("No metrics found at: '%s'\n", dirname);
+			D_GOTO(out, rc = 0);
+		}
+	}
+
+	if (format == D_TM_CSV)
+		d_tm_print_field_descriptors(extra_descriptors, (FILE *)arg);
+
+	while ((num_iter == 0) || (iteration < num_iter)) {
+		d_tm_iterate(ctx, root, 0, filter, NULL, format, extra_descriptors,
+			     iter_cb, arg);
+		iteration++;
+		sleep(delay);
+		if (format == D_TM_STANDARD)
+			printf("\n\n");
+	}
+
+out:
+	if (ctx != NULL)
+		d_tm_close(&ctx);
+	return rc;
+}
+
+static void
+iter_print(struct d_tm_context *ctx, struct d_tm_node_t *node, int level,
+	   char *path, int format, int opt_fields, void *arg)
+{
+	d_tm_print_node(ctx, node, level, path, format, opt_fields, (FILE *)arg);
+}
+
+static void
+iter_reset(struct d_tm_context *ctx, struct d_tm_node_t *node, int level,
+	   char *path, int format, int opt_fields, void *arg)
+{
+	d_tm_reset_node(ctx, node, level, path, format, opt_fields, (FILE *)arg);
+}
```

Comment:
I'm not sure about letting people do this with this tool. Things could get very ugly if another process is actively writing.

Reply:
hmm, the user needs to be able to reset the value.